qa/rpc-tests/test_framework/zip244.py

```diff
@@ -125,8 +125,8 @@ def sapling_outputs_noncompact_digest(saplingBundle):
     digest = blake2b(digest_size=32, person=b'ZTxIdSOutN__Hash')
     for desc in saplingBundle.outputs:
         digest.update(ser_uint256(desc.cv))
-        digest.update(desc.encCiphertext[564:])
-        digest.update(desc.outCipherText)
+        digest.update(desc.encCiphertext[52+564:])
```

Comment:
I'm not sure about this change, but the old code definitely looks wrong. I couldn't find anything in the spec or protocol that describes this. Please check this.

Reply:
That's specified here: https://zips.z.cash/zip-0244#t-3b-iii-sapling-outputs-noncompact-digest

So the code conforms to the spec... going looking for the specification for encCiphertext now.

Reply:
The old code should not be wrong. The first 52 bytes are the note plaintext, the next 512 bytes are the memo. So the remaining data is bytes `(52 + 512):580` or `564:`, i.e. the last 16 bytes (the AEAD tag).